src/pages/studyView/tabs/SummaryTab.tsx

```diff
@@ -163,355 +188,424 @@ export class StudySummaryTab extends React.Component<
                 );
             },
         };
-    }
-
-    renderAttributeChart = (chartMeta: ChartMeta) => {
-        const props: Partial<IChartContainerProps> = {
-            chartMeta: chartMeta,
-            chartType: this.props.store.chartsType.get(chartMeta.uniqueKey),
-            store: this.props.store,
-            dimension: this.store.chartsDimension.get(chartMeta.uniqueKey),
-            title: chartMeta.displayName,
-            filters: [],
-            onDeleteChart: this.handlers.onDeleteChart,
-            isNewlyAdded: this.handlers.isNewlyAdded,
-            studyViewFilters: this.store.filters,
-            analysisGroupsSettings: this.store.analysisGroupsSettings,
-            cancerGeneFilterEnabled: this.store.oncokbCancerGeneFilterEnabled,
-            setComparisonConfirmationModal: this.store
-                .setComparisonConfirmationModal,
-        };
 
-        switch (this.store.chartsType.get(chartMeta.uniqueKey)) {
-            case ChartTypeEnum.PIE_CHART: {
-                //if the chart is one of the custom charts then get the appropriate promise
-                if (
-                    this.store.isUserDefinedCustomDataChart(chartMeta.uniqueKey)
-                ) {
-                    props.filters = this.store
+        this.chartTypeConfig = (
+            chartMeta: ChartMeta,
+            props: Partial<IChartContainerProps>
+        ) => ({
+            [ChartTypeEnum.PIE_CHART]: () => ({
```

Comment:
what's reason for putting these isnide of callbacks?

Reply:
I'm trying to make sure every `subTypeProps` can be created successfully and used consistently.
```
const subTypeProps = this.chartTypeConfig(chartMeta, props)[
    chartType
]();
```
Some of the subtype needs callback to enable `this.store.getMutationDataFiltersByUniqueKey(
                    chartMeta.uniqueKey
                )` to evaluate to some values. Otherwise, it would be an undefined/error.
```
[ChartTypeEnum.MUTATION_EVENT_TYPE_COUNTS_TABLE]: () => ({
    filters: this.store.getMutationDataFiltersByUniqueKey(
        chartMeta.uniqueKey
    ),
    promise: this.store.getMutationEventChartDataCount(chartMeta),
    onValueSelection: this.handlers.onAddMutationDataValues,
    onResetSelection: this.handlers.onSetMutationDataValues,
    id: 'mutation-type-counts-table',
    title: this.store.getChartTitle(
        ChartTypeEnum.MUTATION_EVENT_TYPE_COUNTS_TABLE,
        props.title
    ),
    getData: () =>
        getMutationTypesDownloadData(
            this.store.getMutationEventChartDataCount(chartMeta)
        ),
    downloadTypes: ['Data'],
    onChangeCancerGeneFilter: this.store
        .updateMutatedGenesTableByCancerGenesFilter,
}),
```